File: 🟡 Medium/1497 - Check if Array Pairs Are Divisible by k/_1497CheckIfArrayPairsAreDivisiblebyk.cpp
class Solution {
public:
    bool canArrange(vector<int>& arr, int k) {
        vector<int> freq(k);

        for (auto& el : arr) {
            int modulo = ((el % k) + k) % k;
            freq[modulo] += 1;
        }

        if (freq[0] % 2 != 0) {
            return false;
        }

        for (int i = 1; i <= k / 2; i++) {

            if (freq[i] != freq[k - i]) {
                return false;
            }
        }

        return true;
    }
};

File: 🟡 Medium/1497 - Check if Array Pairs Are Divisible by k/_1497CheckIfArrayPairsAreDivisiblebyk.py
class Solution:
    def canArrange(self, arr: List[int], k: int) -> bool:
        freq = [0] * k

        for el in arr:
            modulo = ((el % k) + k) % k
            freq[modulo] += 1

        if freq[0] % 2 != 0:
            return False

        for i in range(1, (k // 2) + 1):
            if freq[i] != freq[k - i]:
                return False

        return True
